src/ReservoirPair.sol

```diff
@@ -357,32 +361,32 @@ abstract contract ReservoirPair is ReservoirERC20 {
         assetManager.afterLiquidityEvent();
     }
 
-    function adjustManagement(int256 token0Change, int256 token1Change) external {
+    function adjustManagement(int256 aToken0Change, int256 aToken1Change) external {
         require(msg.sender == address(assetManager), "AMP: AUTH_NOT_MANAGER");
-        require(token0Change != type(int256).min && token1Change != type(int256).min, "AMP: CAST_WOULD_OVERFLOW");
+        require(aToken0Change != type(int256).min && aToken1Change != type(int256).min, "AMP: CAST_WOULD_OVERFLOW");
 
-        if (token0Change > 0) {
-            uint104 lDelta = uint104(uint256(int256(token0Change)));
+        if (aToken0Change > 0) {
+            uint104 lDelta = uint104(uint256(aToken0Change));
             token0Managed += lDelta;
             token0.transfer(msg.sender, lDelta);
-        } else if (token0Change < 0) {
-            uint104 lDelta = uint104(uint256(int256(-token0Change)));
+        } else if (aToken0Change < 0) {
+            uint104 lDelta = uint104(uint256(-aToken0Change));
```

Comment:
Albeit there is no guarantee that the `aToken0/1Change` would be <= uint104, I think it would be reasonable to assume that they would be, since it is delta, and not the entire amount. 

What do you think? Do we need a safecast here as well?

Reply:
Delta is almost as big as entire amount, 50% of uint128 is still > uint127, (remember each bit of data doubles/halves the number)

Reply:
mmm yeah I'm keenly aware of that. So I guess a safecast is warranted here?

Reply:
Yeah

Reply:
okay implemented with tests